### PR TITLE
Adds snailpeople as a rare genetic meltdown effect

### PR DESCRIPTION
**Pull request by Time-Green**
Meant to be part of #42864 , but got a little big so I'll add it to wichever gets merged last.
[Snailcrawl demonstration](https://youtu.be/IL7WFpfRo4c)

PROS

- Gets snailcrawl, wich makes them greatly faster and gives them a lube trail

- Their blood is spacelube

- They get a cool armored nodrop snail shell

- They don't slip

CONS

- Very slow

- Punch is pathetic

- Tttaaalllkkk llliiikkkeee ttthhhiiisss

- Extremely vulnerable to salt

- Can't wear glasses

Adds gastrolisis, wich is pretty much [this](https://www.youtube.com/watch?v=lFbPi8o0OEU) spongebob episode where they slowly turn into snails

You can also get gastrolisis by random_reagent_id, so maintpills and botany

:cl:
add: Adds snailpeople as a rare genetics accident.
sprite: Snailshell sprites by nickvr628
/:cl:
Snailshell sprites by @nickvr628
Permission slip:
image

Why: It's a silly gimmicky race and it's a rare occurence. Also extremely highly requested for some reason

### CODE REVIEW

**Comment by subject217** [comment]
Okay, this is epic. Can we get 15 likes?

**Comment by nicbn** [comment]
If it's a librarian make it globglogabgalab

**Comment by MegaEmpirical** [comment]
Are you able to loot their shells even though they don't drop them?

**Comment by Time-Green** [comment]
Yeah. I changed some stuff with stripcode so the shells get emptied on being stripped

**Comment by drderp3635** [comment]
Its happening

**Comment by dr-bee** [comment]
Time to make lines of salt to piss off the genetic freaks!

**Comment by Time-Green** [comment]
Oh yeah, gotta make sure salt decals work

**Comment by anconfuzedrock** [comment]
one step away from snakepeople

**Comment by OverlyGeneric** [comment]
yogstation stole the fucking shell sprite and make it **donator-only.**

**Comment by Time-Green** [comment]
Technicly I stole it from them, with permission. I’m not a very good spriter

**Comment by optimumtact** [comment]
>You can also get gastrolisis by random_reagent_id, so maintpills and **botany**

absolutely not

**Comment by Time-Green** [comment]
Then absolutely not it is

**Comment by subject217** [comment]
rip magic maint snailpills

**Comment by TimeIsEndless** [comment]
So was that just a no to botany snailpeople or also a no to maint snailpeople?

**Comment by Time-Green** [comment]
I think oranges mainly meant botany snails, but I can’t have one without the other without some form of reworking chem code or just straight up shitcode it

**Comment by TimeIsEndless** [comment]
That's a shame

**Comment by fluffe9911** [comment]
@Time-Green Cant you make a pill that looks like a maint pill that contains gastrolisis then add it to the maint spawn table?

**Comment by Time-Green** [comment]
Then it would be alot more likely to spawn. I am also not going to fuck with the weighted system with decimals. I said this would be a genetics thing, and I don't think I am gonna start actively putting in other ways to achieve it in fear of my snail permission pass being pulled.

**Comment by TimeIsEndless** [comment]
Also if you do this you should make it contain the minimum amount of gastrolisis that is necessary to turn you into a snail so you can't just grind it up and share it.

**Comment by fluffe9911** [comment]
Fair enough I guess

**Comment by Time-Green** [comment]
This is done now.

File: code/game/objects/effects/decals/cleanable/food.dm [hunk 1]
```diff
@@ -30,6 +30,12 @@
 	desc = "A sizable pile of table salt. Someone must be upset."
 	icon_state = "salt_pile"
 
+/obj/effect/decal/cleanable/food/salt/CanPass(atom/movable/AM, turf/target)
+	if(is_species(AM, /datum/species/snail))
+		to_chat(AM, "<span class='danger'>Your path is obstructed by <span class='phobia'>salt</span></span>")
```
Comment (PKPenguin321) [inline code/game/objects/effects/decals/cleanable/food.dm]: punctuation

**Comment by PKPenguin321** [comment]
getting forced on top of salt as a snailperson should apply acid to you or something

**Comment by Nickvr628** [comment]
The slope has been sufficiently slipperyed, my work is almost complete!

File: code/game/objects/effects/decals/cleanable/food.dm [hunk 2]
```diff
@@ -32,10 +32,21 @@
...
 	if(is_species(AM, /datum/species/snail))
-		to_chat(AM, "<span class='danger'>Your path is obstructed by <span class='phobia'>salt</span></span>")
-		return FALSE
+		var/mob/living/carbon/human/H = AM
+		if(H.stat == CONSCIOUS && !H.IsStun())
+			to_chat(H, "<span class='danger'>Your path is obstructed by <span class='phobia'>salt</span>.</span>")
+			return FALSE
 	return TRUE
 
+/obj/effect/decal/cleanable/food/salt/Crossed(atom/movable/AM)//will only really happen when a snail is teleported onto salt
```
Comment (optimumtact) [inline code/game/objects/effects/decals/cleanable/food.dm]: you're going to add proc overhead on every salt cross, for a snail.

no

**Comment by optimumtact** [comment]
only other point of feedback here is that sprite is genuinely awfully bad.

**Comment by Time-Green** [comment]
Snails just can't pass over salt, I got rid of the crossed fuckening.
I am of course a godlike spriter, but snails are supposed to be abominations so I made the sprite intentionally bad

**Comment by datorangebottle** [comment]
Sprite is kinda bad, also we'll basically never see them.

**Comment by optimumtact** [comment]
good

**Comment by subject217** [comment]
It's roundstart race time.

**Comment by optimumtact** [comment]
do not fucking dare, I will burn this repo to the ground

**Comment by subject217** [comment]
Sorry oranges, but roundstart snailpeople are quite simply the People's choice. It's the future we deserve.

**Pull request merged**